src/shogun/machine/GeneralizedLinearMachine.cpp

```diff
@@ -0,0 +1,299 @@
+			break;		
+	}
+
+	//Update the estimated variables
+	m_w = w;
+	m_bias = bias;
+
+	return true;
+}
+
+//TODO
+std::shared_ptr<RegressionLabels> GeneralizedLinearMachine::apply_regression(std::shared_ptr<Features> data)
+{
+	
+}
+
+//TODO
+SGVector<float64_t> GeneralizedLinearMachine::apply_get_outputs(std::shared_ptr<Features> data)
+{
+
+}
+
+//TODO
+bool GeneralizedLinearMachine::train_machine(std::shared_ptr<const DenseFeatures<float64_t>> feats)
+{
+
+}
+
+SGVector<float64_t> GeneralizedLinearMachine::get_w() const
+{
+	return m_w;
+}
+
+void GeneralizedLinearMachine::set_w(const SGVector<float64_t> w)
+{
+	m_w = w;
+}
+
+void GeneralizedLinearMachine::set_bias(float64_t b)
+{
+	m_bias=b;
+}
+
+float64_t GeneralizedLinearMachine::get_bias() const
+{
+ 	return m_bias;
+}
+
+void GeneralizedLinearMachine::set_features(std::shared_ptr<DotFeatures> feat)
+{
+ 	features=std::move(feat);
+}
+
+std::shared_ptr<DotFeatures> GeneralizedLinearMachine::get_features()
+{
+	return features;
+}
+
+SGVector<float64_t> GeneralizedLinearMachine::conditional_intensity(SGMatrix<float64_t> X, SGVector<float64_t> w, float64_t bias)
+{
+	SGVector<float64_t> z = compute_z(X, w, bias);
+	SGVector<float64_t> result = non_linearity(z);
+	return result;
+}
+
+SGVector<float64_t> GeneralizedLinearMachine::compute_z(SGMatrix<float64_t> X, SGVector<float64_t> w, float64_t bias)
+{
+	SGVector<float64_t> z = linalg::matrix_prod(X, w, false);
+	return z;
+}
+
+SGVector<float64_t> GeneralizedLinearMachine::non_linearity(SGVector<float64_t> z)
+{
+	SGVector<float64_t> result;
+	switch (distribution)
+	{
+	case POISSON:
+		result = SGVector<float64_t>(z);
+		float64_t bias = 0;
+
+		if(m_fit_intercept)
+			bias = (1 - m_eta) * exp(m_eta);
+
+		for (int i = 0; i < z.vlen; i++)
+		{
+			if(z[i]>m_eta)
+				result[i] = z[i] * exp(m_eta) + bias;
+			else
+				result[i] = exp(z[i]);
+		}
+		break;
+	
+	default:
+		break;
+	}
+	return result;
+}
+
+SGVector<float64_t> GeneralizedLinearMachine::gradient_non_linearity(SGVector<float64_t> z)
+{
+	SGVector<float64_t> result;
+	switch (distribution)
+	{
+	case POISSON:
+		result = SGVector<float64_t>(z);
+		for (int i = 0; i < z.vlen; i++)
+		{
+			if(z[i]>m_eta)
+				
+				result[i] = exp(m_eta);
+			else
+				result[i] = exp(z[i]);
+		}
+		break;
+	
+	default:
+		break;
+	}
+	return result;
+}
+
+SGVector<float64_t> GeneralizedLinearMachine::compute_grad_L2_loss_w(SGMatrix<float64_t> X, SGVector<float64_t> y, SGVector<float64_t> w, float64_t bias)
+{
+	int32_t n_samples = y.vlen;
+	int32_t n_features = X.num_rows;
+	if(m_tau == NULL)
+		m_tau = SGMatrix<float64_t>::create_identity_matrix(w.vlen, 1.0);
+	SGMatrix<float64_t> inv_cov = linalg::matrix_prod(m_tau, m_tau, true, false);
+	
+	SGVector<float64_t> z = compute_z(X, w, bias);
+	SGVector<float64_t> mu = non_linearity(z);
+	SGVector<float64_t> grad_mu = gradient_non_linearity(z);
+
+	SGVector<float64_t> grad_w(w.vlen);
+	switch (distribution)
+	{
+	case POISSON:
```

Comment:
this could also be grouped with the class I mentioned above, so that everything related to poisson is in one place